### ISSUE TITLE
How can I shorten time of make band structure?

### ISSUE THREAD

**Issue by 91bsjun** [materialsproject/pymatgen#251]
--- CODE ---
run = Vasprun(pwd+"/Band-DOS/vasprun.xml",parse_projected_eigen = True)
bands = run.get_band_structure(pwd+"/Band-DOS/KPOINTS",efermi = dosrun.efermi,line_mode = True)
plotter = BSPlotterProjected(bands)
graph = plotter.get_elt_projected_plots_color()
--- CODE end ---

when I use small vasprun.xml (500KB~1MB) , It takes few seconds. 
But If size of vasprun.xml file is big (about 20MB ?), it takes so long time.. 3~4 minutes
How can I shorten time?? Can pymatgen use python multiprocessing or Parallel Python ??
Thanks

**Comment by gVallverdu** [comment]
Hi
I got the same issues with large numbers of Kpoints or huge vasprun.xml files (several hundred of Mo). I guess there are too many xml tags/attribute in the file. In that cases I used Poscar class instead. I am working on it because, for the moment, that class does not return an electronic_strucure.Band object. Moreover there is a bug because in the case of spin polarized calculations, because only beta bands are returned.

I will push a commit about that quickly. I wrote the code but I need time to check the band structure object and compare two objects extracted from Vasprun or Procar class.
